independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientResponseCompleteRestHandler.java

```diff
@@ -64,11 +64,19 @@ public static ResponseImpl mapToResponse(RestClientRequestContext context,
                 Object result = multipartData.newInstance();
                 builder.entity(result);
                 List<InterfaceHttpData> parts = context.getResponseMultipartParts();
-                for (FieldFiller fieldFiller : multipartData.getFieldFillers()) {
-                    InterfaceHttpData httpData = getPartForName(parts, fieldFiller.getPartName());
-                    if (httpData == null) {
+                for (InterfaceHttpData httpData : parts) {
+                    FieldFiller fieldFiller = null;
+                    // find the correct filler
+                    for (FieldFiller ff : multipartData.getFieldFillers()) {
+                        if (ff.getPartName().equals(httpData.getName())) {
+                            fieldFiller = ff;
+                            break;
+                        }
+                    }
+                    if (fieldFiller == null) {
                         continue;
-                    } else if (httpData instanceof Attribute) {
+                    }
+                    if (httpData instanceof Attribute) {
```

Comment:
No biggie, but this can be done here: 
```suggestion
                    if (httpData instanceof Attribute att) {
```